pkg/collector/corechecks/snmp/internal/checkconfig/config_metric.go

```diff
@@ -84,6 +84,8 @@ type MetricsConfig struct {
 
 	ForcedType string              `yaml:"forced_type"`
 	Options    MetricsConfigOption `yaml:"options"`
+
+	ScaleFactor float64 `yaml:"scale_factor"`
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** (Sorry for commenting late on this) I think it can help to move `scale_factor` to `SymbolConfig`: https://github.com/DataDog/datadog-agent/blob/6e8e1a93a32da5a900448a80fdb1f29a00410100/pkg/collector/corechecks/snmp/internal/checkconfig/config_metric.go#L18-L28

That way, when scale_factor is used with table/column definition, it can be applied to individual symbol instead of all symbols/metrics of the table.‏

Reply:
The signature for `sendMetric` would require one additional parameter, making it bit long (6 parameters), which is not ideal. Moving everything to `SymbolConfig` is out of the scope of this PR, will see what I can do as a best effort